pkg/cloud/openstack/deployer.go

```diff
@@ -62,15 +65,28 @@ func (d *DeploymentClient) GetKubeConfig(cluster *clusterv1.Cluster, master *clu
 		return "", fmt.Errorf("unable to use HOME environment variable to find SSH key: %v", err)
 	}
 
+	machineConfig, err := d.providerconfig(master.Spec.ProviderConfig)
+	if err != nil {
+		return "", err
+	}
+
 	result := strings.TrimSpace(util.ExecCommand(
 		"ssh", "-i", homeDir+"/.ssh/openstack_tmp",
 		"-o", "StrictHostKeyChecking no",
 		"-o", "UserKnownHostsFile /dev/null",
-		fmt.Sprintf("%s@%s", SshUserName, ip),
+		fmt.Sprintf("%s@%s", machineConfig.SshUserName, ip),
 		"echo STARTFILE; sudo cat /etc/kubernetes/admin.conf"))
 	parts := strings.Split(result, "STARTFILE")
 	if len(parts) != 2 {
 		return "", nil
 	}
 	return strings.TrimSpace(parts[1]), nil
 }
+
+func (d *DeploymentClient) providerconfig(providerConfig clusterv1.ProviderConfig) (*openstackconfigv1.OpenstackProviderConfig, error) {
```

Comment:
What does this func have to do with the `*DeploymentClient`? Seems like this shouldn't belong on that struct

Reply:
kinda, sorta, maybe! It's a utility to unmarshal the providerConfig raw data into a providerConfig struct. I'd agree this is prob not the best place to put it but I thought we could keep it here while I work on cleaning up other parts of the code.

I copied this function from the OpenStackClient implementation, which makes it even worse as I basically duplicated code :P it's just temporary, tho.

Reply:
Since it doesn't require `*DeploymentClient` for input, nor does it mutate `*DeploymentClient` in any way it is much better to leave this func unattached to a struct, or better to move into the `openstackproviderconfig` package to give a consistent func for unmarshalling

Reply:
Yes, as I said earlier. I don't this function belongs to this struct but, I'm trying to take baby steps and introduce small refactors as we don't have good test coverage yet.

Small refactor -> let ppl test things -> repeat :smile: